### PR TITLE
DLS-8434 added navigation for select change and error handling, cacheing on calls

### PR DESCRIPTION
**Pull request by Chrissystk**
https://github.com/hmrc/soft-drinks-industry-levy-ui-tests/pull/148

### CODE REVIEW

File: app/models/updateRegisteredDetails/UpdateContactDetails.scala [hunk 1]
```diff
@@ -16,10 +16,20 @@
 
 package models.updateRegisteredDetails
 
+import models.Contact
 import play.api.libs.json._
 
 case class UpdateContactDetails (fullName: String, position: String, phoneNumber: String, email: String)
```
Comment (RaminEsfandiari) [inline app/models/updateRegisteredDetails/UpdateContactDetails.scala:22]: I think ContactDetails would be a more suitable name for this model.

**Review by RaminEsfandiari — CHANGES_REQUESTED** [review]

File: app/controllers/SelectChangeController.scala [hunk 2]
```diff
@@ -44,42 +45,44 @@ class SelectChangeController @Inject()(
...
-      .map {
-        case (site, index) => (index.toString, Warehouse.fromSite(site))
-      }.toMap[String, Warehouse]
-    UserAnswers(id = request.sdilEnrolment, journeyType = value,contactAddress = request.subscription.address, packagingSiteList = currentPackagingSites, warehouseList = currentWarehouses)
+  private def getRedirectUrl(value: SelectChange): Call = {
+    value match {
+      case SelectChange.ChangeActivity => changeActivity.routes.AmountProducedController.onPageLoad(NormalMode)
+      case SelectChange.CorrectReturn => correctReturn.routes.SelectController.onPageLoad(NormalMode)
+      case SelectChange.CancelRegistration => cancelRegistration.routes.ReasonController.onPageLoad(NormalMode)
+      case _ => routes.IndexController.onPageLoad
```
[the remark this answers is not in the document]
Comment (Chrissystk) [reply app/controllers/SelectChangeController.scala:84]: Yes I will add a note on the ticket to do that

File: app/models/updateRegisteredDetails/UpdateContactDetails.scala (hunk 1 above)
Comment (Chrissystk) [reply app/models/updateRegisteredDetails/UpdateContactDetails.scala:22]: Contact is what it is called across all the services it is part of the RetrievedSubscription model.

Comment (RaminEsfandiari) [reply app/models/updateRegisteredDetails/UpdateContactDetails.scala:22]: Not sure why it tagged line 19. I was referring to UpdateContactDetails as an unsuitable name.

**Pull request merged**